src/common/decorators/swagger.ts

```diff
@@ -115,22 +115,39 @@ export function ApiTransactionResponse(
   );
 }
 
+type FailedTransactionResponseCodes =
+  | HttpStatus.NOT_FOUND
+  | HttpStatus.BAD_REQUEST
+  | HttpStatus.UNPROCESSABLE_ENTITY;
+
 /**
- * A helper that combines responses for SDK Error like `BadRequestException`, `NotFoundException`, `UnprocessableEntityException`, `InternalServerErrorException`
+ * A helper that combines responses for SDK Errors like `BadRequestException`, `NotFoundException`, `UnprocessableEntityException`
  *
- * @param options - these will be passed to the `ApiNotFoundResponse` decorator
+ * @param messages - key value map of API response descriptions that will be passed to appropriate `MethodDecorator`
  */
 export function ApiTransactionFailedResponse(
-  options: ApiResponseOptions
+  messages: Partial<Record<FailedTransactionResponseCodes, string | undefined>>
 ): ReturnType<typeof applyDecorators> {
-  return applyDecorators(
-    ApiNotFoundResponse(options),
-    ApiBadRequestResponse({
-      description: 'Returned if transaction validation failed or same data already exists in chain',
-    }),
-    ApiUnprocessableEntityResponse({
-      description:
-        'Returned if there is insufficient balance to perform action or required perquisites for transaction have not been met',
-    })
-  );
+  const decoratorMap: Record<
+    FailedTransactionResponseCodes,
+    (description?: string) => MethodDecorator
+  > = {
+    [HttpStatus.NOT_FOUND]: (description = 'Entity was not found') =>
+      ApiNotFoundResponse({ description }),
+    [HttpStatus.BAD_REQUEST]: (
+      description = 'Returned if transaction validation failed or same data already exists in chain'
+    ) => ApiBadRequestResponse({ description }),
+    [HttpStatus.UNPROCESSABLE_ENTITY]: (
+      description = 'Returned if there is insufficient balance to perform action or required perquisites for transaction have not been met'
+    ) => ApiUnprocessableEntityResponse({ description }),
+  };
+  const decorators: MethodDecorator[] = [];
+
+  for (const [key, value] of Object.entries(messages)) {
+    const decorator = decoratorMap[key as unknown as FailedTransactionResponseCodes];
```

Comment:
Try to avoid casting to unknown like this, functionally its `key as any`.

IDK how to stop TS from striping the types from the whole string vs number index is technically always number.

`const keyAsCode = Number(key) as SupportedHttpStatusCodes;` I'd do something like this, or even write a typeguard function.

There might be a full TS way to do this but I don't know it.